ocaml/xapi/helpers.ml

```diff
@@ -43,6 +43,24 @@ let log_exn_continue msg f x =
 
 type log_output = Always | Never | On_failure
 
+let filter_args args =
+  let patterns =
+    ["^\\(.*proxy_username=\\)\\(.*\\)$"; "^\\(.*proxy_password=\\)\\(.*\\)$"]
```

Comment:
These strings are static and thus `Re.Str.regexp` should be called only once for them and not at every invocation of `filter_args`. I also think this should be one pattern and not multiple such that the string being filtered is only scanned once.

Reply:
This can be slightly simplified by using `{| .. |}` for the string because`\` does not need to be escaped.